mobile/Ihc/components/UpdateMedicationModal.js

```diff
@@ -19,41 +19,70 @@ export default class UpdateMedicationModal extends Component<{}> {
    *    showModal: boolean
    *    closeModal: function
    *    saveModal: function
-   *    updateMedication: function
-   *    medicationToEdit: Medication object
+   *    formValues: formValues
    *  }
    */
   constructor(props) {
     super(props);
   }
+  
+  /*MedicationInventoryForm = t.struct({
+    drugName: t.String, // drug name
+    quantity: t.int,
+    dosage: t.int,
+    units: t.String,
+    comments: t.maybe(t.String)
+  });*/
 
-  updateName(drugName) {
-    this.props.medicationToEdit.drugName = drugName;
-    this.props.updateMedication(this.medication);
-  }
+  Units = t.enums({
+    kg: 'kg',
+    ml: 'ml',
+    g: 'g'
+  });
 
-  updateQuantity(quantity) {
-    this.props.medicationToEdit.quantity = parseInt(quantity,10);
-    this.props.updateMedication(this.medication);
-  }
+  Medication = t.struct({
+    drugName: t.String,
+    quantity: t.Number,
+    dosage: t.Number,
+    units: this.Units,
+    comments: t.maybe(t.String)
+  });
 
-  updateDosage(dosage) {
-    this.props.medicationToEdit.dosage = parseInt(dosage, 10);
-    this.props.updateMedication(this.medication);
+  formOptions = {
+    fields: {
+      drugName: {
```

Comment:
I think multiline is false by default. If so, removing this would be cleaner, but that's not a big deal if you dont